### DISCUSSION TITLE
Docs: Working Locally With gRPC Services

### DISCUSSION THREAD

**Discussion by towens** [istio/istio#52392] [Q&A] [answered]
Hello,

I am looking to work with gRPC services locally and having some trouble traffic routing as opposed to HTTP services.

I'd be happy to write something up, if there is an equivalent to `1. Access the httpbin service using curl:` under: https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/#accessing-ingress-services

Where you are setting the host header for the gateway to pick up.

Local environment:
- Mac OSX.
- Colima running a Kind cluster.
- Istio deployed to Kind.
- Metallb.
- Port forwarding the ingress-gateway svc.

So it seems the issue is that HTTP/2 is using `authority` not `host` and without setting:
```
    hosts:
      - "*"
```
In the gw and vs. The request never hits the gRPC svc. Clearly not an issue on production with a proper host. I just really love how fast it is to iterate locally and how much money it doesn't cost in a test cluster in the ☁️ .

Thanks for any guidance and apologies if I've missed a doc covering this.

**Comment by howardjohn** [comment]
>  HTTP/2 is using authority not host and without setting:

`host` and `authority` are synonyms for the same thing, at least in the context of Istio, so I don't think that is exactly the issue.

What GRPC client are you using? That is where you would configure the host/authority. Of course, you can also use `*` so it doesn't matter

**Reply by towens** [reply]
Presently using grpcurl and insomnia.

Setting the host header never shows up in the logs, debug is enabled. Setting authority causes it to show up twice. Once for `localhost:port-forwarded-ingress-svc-port` and the one I set `myservice.local.com`.

Yes, it is working with `*` and only `*`. We don't have anything in prod with `*` and this is only thing locally that isn't working like one of our production clusters.

If I'm chasing a ghost locally it's no problem,  I can put it down and rely on `*` when working with gRPC locally. Just making sure I didn't miss the one thing causing it to fail. 

Istio info as it is not listed above.
```
client version: 1.21.2
control plane version: 1.21.2
data plane version: 1.21.2
```

**Comment by towens** [comment, marked as answer]
The most basic client implementation solves the issue. 

```
ktl -n istio-system port-forward svc/istio-ingressgateway 61778:80
Forwarding from 127.0.0.1:61778 -> 8080
Forwarding from [::1]:61778 -> 8080
```
The Istio gw, vs list `myservice.local.com as a host.

```
	conn, err := grpc.Dial("localhost:61778",
		grpc.WithTransportCredentials(insecure.NewCredentials()),
		grpc.WithAuthority("myservice.local.com"))
```


Thanks.
